src/metadata/metadata_handler.cc

```diff
@@ -201,42 +218,48 @@ std::string MetadataHandler::getMimeType() const
 }
 
 static constexpr std::array chKeys = {
-    std::pair(CH_DEFAULT, "Default"),
+    std::pair(ContentHandler::UNKNOWN, "Unknown"),
+    std::pair(ContentHandler::DEFAULT, "Default"),
 #ifdef HAVE_LIBEXIF
-    std::pair(CH_LIBEXIF, "LibExif"),
+    std::pair(ContentHandler::LIBEXIF, "LibExif"),
 #endif
 #ifdef HAVE_TAGLIB
-    std::pair(CH_ID3, "TagLib"),
+    std::pair(ContentHandler::ID3, "TagLib"),
 #endif
-    std::pair(CH_TRANSCODE, "Transcode"),
-    std::pair(CH_EXTURL, "Exturl"),
-    std::pair(CH_MP4, "MP4"),
+    std::pair(ContentHandler::TRANSCODE, "Transcode"),
+    std::pair(ContentHandler::EXTURL, "Exturl"),
+    std::pair(ContentHandler::MP4, "MP4"),
 #ifdef HAVE_FFMPEGTHUMBNAILER
-    std::pair(CH_FFTH, "FFmpegThumbnailer"),
+    std::pair(ContentHandler::FFTH, "FFmpegThumbnailer"),
 #endif
-    std::pair(CH_FLAC, "Flac"),
-    std::pair(CH_FANART, "Fanart"),
-    std::pair(CH_CONTAINERART, "ContainerArt"),
+    std::pair(ContentHandler::FLAC, "Flac"),
+    std::pair(ContentHandler::FANART, "Fanart"),
+    std::pair(ContentHandler::CONTAINERART, "ContainerArt"),
 #ifdef HAVE_MATROSKA
-    std::pair(CH_MATROSKA, "Matroska"),
+    std::pair(ContentHandler::MATROSKA, "Matroska"),
 #endif
 #ifdef HAVE_WAVPACK
-    std::pair(CH_WAVPACK, "WavPack"),
+    std::pair(ContentHandler::WAVPACK, "WavPack"),
 #endif
-    std::pair(CH_SUBTITLE, "Subtitle"),
-    std::pair(CH_RESOURCE, "Resource"),
+    std::pair(ContentHandler::SUBTITLE, "Subtitle"),
+    std::pair(ContentHandler::RESOURCE, "Resource"),
 };
 
-int MetadataHandler::remapContentHandler(const std::string& contHandler)
+ContentHandler MetadataHandler::remapContentHandler(const std::string& contHandler)
 {
     auto chEntry = std::find_if(chKeys.begin(), chKeys.end(), [contHandler](auto&& entry) { return contHandler == entry.second; });
     if (chEntry != chKeys.end()) {
         return chEntry->first;
     }
-    return -1;
+    return ContentHandler::UNKNOWN;
```

Comment:
I think it would be better to make this method throw and not have the unknown value.

Reply:
This is one reason for it: https://github.com/KarlStraussberger/gerbera/blob/c419bb294f3260742f76ee8376f2910e6f52be7b/src/upnp_xml.cc#L55
We do not have a check for the config yet.

And the extra check for https://github.com/KarlStraussberger/gerbera/blob/c419bb294f3260742f76ee8376f2910e6f52be7b/src/cds_resource.h#L120-L123. This is easy to fix.

Reply:
re `isMetaResource` IMO we should get rid of that method (not in this change)

My thinking is that we should add a Purpose enum to Resource too:

`Purpose { Content, Thumbnail, Subtitle }`

Then you can tell the purpose of the resource explicitly.

Reply:
And for the other `CFG_IMPORT_RESOURCES_ORDER` it simply skips it if its not mapped anyway, which could be as handled with a try/catch.

Or if you wanted it to be explicit it should return an optional.

Reply:
> And for the other `CFG_IMPORT_RESOURCES_ORDER` it simply skips it if its not mapped anyway, which could be as handled with a try/catch.
> 
> Or if you wanted it to be explicit it could return an optional.

Moved that into the check function of the config value. So it throws on startup now